### PR TITLE
adds /api/tag/{tagKey}

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: test/test.tags.js [hunk 1]
```diff
@@ -134,3 +134,62 @@ lab.experiment('tags', { timeout: 5000 }, () => {
...
+      done();
+    });
+  });
+  lab.test('can use the /api/tag/{tagKey} route to get a list of distinct keys', { timeout: 5000 }, (done) => {
+    server.inject({
+      url: '/api/tag/currency',
+      method: 'GET'
+    }, (response) => {
+      code.expect(response.statusCode).to.equal(200);
+      code.expect(response.result.length).to.equal(2);
```
Comment (jgallen23) [inline test/test.tags.js]: shouldn't this be 3? hi, dollars and yen

File: routes/api/tag.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+exports.tag = {
+  method: 'GET',
+  path: '/api/tag/{tagKey}',
+  handler(request, reply) {
+    const filter = {};
```
Comment (jgallen23) [inline routes/api/tag.js]: can we add support for `type` filtering too?

File: routes/api/tag.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+exports.tag = {
+  method: 'GET',
+  path: '/api/tag/{tagKey}',
```
Comment (jgallen23) [inline routes/api/tag.js]: lets change this to be this structure: `/api/tag-values?tag={value}`

File: routes/api/tag.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  path: '/api/tag-values',
+  handler(request, reply) {
+    const filter = {};
+    filter[`tags.${request.query.tag}`] = { '$exists': 1 };
+    if (request.query.type) {
+      filter.type = request.query.type;
+    }
+    request.server.methods.tag(request.query.tag, filter, (err, results) => {
+      if (err) {
+        return reply('Error').code(500);
```
Comment (jgallen23) [inline routes/api/tag.js]: just `reply(err)`

File: test/test.tags.js [hunk 5]
```diff
@@ -134,3 +134,73 @@ lab.experiment('tags', { timeout: 5000 }, () => {
...
+
+lab.experiment('tag', { timeout: 5000 }, () => {
+  let server;
+  lab.beforeEach({ timeout: 5000 }, (done) => {
+    setup({}, (err, result) => {
+      if (err) {
+        return done(err);
+      }
+      server = result;
+      server.plugins['hapi-mongodb'].db.collection('tracks').drop();
```
Comment (jgallen23) [inline test/test.tags.js]: seems like we do this for all tests. can we reuse code at all?

**Comment by orthagonal** [comment]
Gonna do one more refactor to improve the tests before I submit this for review again.

File: test/setup.test.js [hunk 6]
```diff
@@ -3,10 +3,25 @@
 // for unit testing outside of docker
 const Rapptor = require('rapptor');
 
-module.exports.withRapptor = (options, callback) => {
+module.exports.server = undefined;
+module.exports.withRapptor = (options, dataSet, callback) => {
```
Comment (jgallen23) [inline test/setup.test.js]: take a look at this: https://github.com/firstandthird/inspired-survey/blob/master/test/setup.test.js

**Pull request merged**